packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -152,6 +152,7 @@ export class GraphQLSitemapService {
 
       const items = data?.search.results ? data.search.results : [];
 
+      // todo: convert to array.map (Anastasiya, March 2021)
       const staticPaths = items.reduce((list: StaticPath[], item: { url: { path: string } }) => {
```

Comment:
@anastasiya29 Why can't we use `reduce`? :)

Reply:
Two reasons.
1. Using `map`  would be more efficient. The reason is that only 1 new array would be created. Right now, a new array is created for every array entry that it iterates over (because of the `Array.concat`).
2.  Using `map` would make the intention of the code easier to understand. What the code is doing here is updating every single value in the array, without changing the size of the array. This is exactly what `map` is for. `reduce` is usually used when you want to transform an array to a new type, like an Object.

Reply:
Interesting. Thank you for explanation, I didn't think about it